src/newmatrix.cpp

```diff
@@ -2983,7 +2983,11 @@ void process_upload(struct matrix_icon *persona)
 {
   // Note: We only upload one file at a time. Find the first valid one and process it, then stop.
   if (persona && persona->decker && persona->decker->deck) {
-    for (struct obj_data *soft = persona->decker->deck->contains, *next_obj; soft; soft = next_obj) {
+    obj_data *deck = persona->decker->deck;
+    if (persona->decker->proxy_deck)
+      deck = persona->decker->proxy_deck;
```

Comment:
If you're making this change, you should propagate it to the rest of this block. There's a call to change storage memory below that still hits the original deck.

Reply:
I even read the code and still didn't see that. I am just getting old and my eyesight is going, obviously.